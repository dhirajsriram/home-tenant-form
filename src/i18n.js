import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { initReactI18next } from "react-i18next";

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    // we init with resources
    resources: {
      de: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Gehalt",
          "Review": "Rezension",
          "Enter your name": "Gib deinen Namen ein",
          "First Name": "Vorname",
          "First name": "Vorname",
          "Last Name": "Nachname",
          "Last name": "Nachname",
          "Enter your Email Address": "Geben sie ihre E-Mail Adresse ein",
          "Enter your phone": "Geben Sie Ihr Telefon ein",
          "Enter your Salary": "Geben Sie Ihr Gehalt ein",
          "Kindly review your input": "Bitte überprüfen Sie Ihre Eingabe",
          "Over 4000": "Mehs als 4000",
          "submit":"Registrieren",
          next: "weiter",
          back: "Bisherige"
        },
      },
      en: {
        translations: {
          "Name": "Name",
          "Email":"Email",
          "Phone": "Phone",
          "Salary": "Salary",
          "Review": "Review", 
          "Enter your name": "Enter your name",
          "First Name": "First Name",
          "First name": "First Name",
          "Last Name": "Last Name",
          "Last name": "Last Name",
          "Enter your Email Address": "Enter your Email Address",
          "Enter your phone": "Enter your Email Phone",
          "Enter your Salary": "Enter your Email Salary",
          "Kindly review your input":"Kindly review your input",
          "Over 4000": "Over 4000",
          "submit":"submit",
          next: "next",
          back: "back"
        },
      },
    },
    fallbackLng: "en",
    debug: true,

    // have a common namespace used around the full app
    ns: ["translations"],
    defaultNS: "translations",

    keySeparator: false, // we use content as keys

    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
