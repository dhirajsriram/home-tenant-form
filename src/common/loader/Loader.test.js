import React from "react";
import Loader from "./Loader";
import Adapter from "enzyme-adapter-react-16";
import { configure, mount } from "enzyme";
import "../../i18n";
import { MemoryRouter } from "react-router";
import { Provider } from "react-redux";
import { createStore } from "redux";
import allReducers from "../../store/reducer";

let wrapper;
configure({ adapter: new Adapter() });

const store = createStore(allReducers);

beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <React.Suspense fallback={<div></div>}>
        <MemoryRouter initialEntries={["/"]} keyLength={0}>
          <Loader />
        </MemoryRouter>
      </React.Suspense>
    </Provider>
  );
});

describe("<Loader>", () => {
  it("Renders without exploding", async () => {
    expect(wrapper).toMatchSnapshot();
  });
});
