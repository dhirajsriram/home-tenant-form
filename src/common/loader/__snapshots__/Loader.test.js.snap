// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> Renders without exploding 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Suspense
    fallback={<div />}
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/",
        ]
      }
      keyLength={0}
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Loader>
          <div
            className="loader"
          />
        </Loader>
      </Router>
    </MemoryRouter>
  </Suspense>
</Provider>
`;
